File: fnc/examples/chapter09/chebPhi.m
%%
x = linspace(-1,1,1601)';
Phi = zeros(size(x));
for n = 10:10:50
    theta = linspace(0,pi,n+1)';
    t = -cos(theta);                    
    for k = 1:length(x)
        Phi(k) = prod(x(k)-t);
    end
    semilogy(x,abs(Phi))
    hold on
end
axis tight, title('Effect of Chebyshev nodes')    % ignore this line
xlabel('x'), ylabel('|\Phi(x)|')   % ignore this line
ylim([1e-18 1e-2])   % ignore this line

%%
% The convergence is a bit slower in the middle than with equally spaced
% points, but far more uniform over the entire interval, which is the key
% to global convergence. 

File: julia/scripts/chapter09/chebPhi.jl

using Plots
plot([],[],xaxis=("\$x\$"),yaxis=(:log10,"\$|\\Phi(x)|\$",[1e-18,1e-2]))

x = range(-1,stop=1,length=1601)
for n = 10:10:50
    theta = pi*(0:n)/n
    t = @. -cos(theta)                   
    Phi = [ prod(xk.-t) for xk in x ]
    plot!(x,abs.(Phi),m=(1,:o,stroke(0)))
end
plot!([],[],legend=:none,title="Effect of Chebyshev nodes")
